plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginCredentialExporter.java

```diff
@@ -0,0 +1,48 @@
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jface.window.IShellProvider;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.cloud.tools.eclipse.appengine.login.GoogleLoginService;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+
+public class LoginCredentialExporter {
+
+  private static final String CREDENTIAL_FILENAME = "gcloud-credentials.json";
+  
+  private GoogleLoginService loginService;
+  
+  public LoginCredentialExporter() {
+    this(new GoogleLoginService());
+  }
+  
+  @VisibleForTesting
+  LoginCredentialExporter(GoogleLoginService googleLoginService) {
+    this.loginService = googleLoginService;
+  }
+
+  public void logInAndSaveCredential(IPath workDirectory, IShellProvider shellProvider) throws IOException, 
+                                                                                               CoreException {
+    Credential credential = loginService.getActiveCredential(shellProvider);
+    if (credential == null) {
+      throw new CoreException(StatusUtil.error(LoginCredentialExporter.class, "Login failed"));
```

Comment:
I'm just thinking if it makes sense for a helper class in the `.login.ui` bundle to throw `CoreException` and why it has to be `CoreException`. Would be nice if this helper class is decoupled with this Eclipse specific exception.

Perhaps, it's just enough to make this method accept a Credential and just write it to a file rather than inducing the login process? I don't know.